### ISSUE TITLE
Unable to load on the Libra 2

### ISSUE THREAD

**Issue by imoBooze** [Aryetis/Obenkyobo#2]
Every time I attempted to start Obenkyobo, it seems to hang on the welcome popup, which I am unable to click away.  NickleMenu seems to be working fine, as I installed KOReader previously.  My device is the Kobo Libra 2.
[log.txt](https://github.com/Aryetis/Obenkyobo/files/11790603/log.txt)

**Comment by Aryetis** [comment]
Hello, 
Thanks for trying Obenkyobo out. Sorry about the bug, I don't have a libra 2 to test it on but we can try to figure things out.
1/ Is the log.txt really empty ?
2/ When you mean "hang on the welcome popup". What do you mean exactly ? Does it show you the "This software is still in development, etc"  poup and you can't click on "ok" ?
3/ Can you try to edit the file in `.adds\Obenkyobo\Obenkyobo_launcher.sh` find the `export QT_QPA_PLATFORM=${QTPLUGINOBENKYOBO}:` line and turn it into `export QT_QPA_PLATFORM=${QTPLUGINOBENKYOBO}:experimentaltouchhandler`
4/ What version of Obenkyobo did you install ?

**Comment by imoBooze** [comment]
Hi,
1. Huh, something must've gone wrong while uploading it to Github, it shouldn't be empty.  I'll try uploading again.
[log.txt](https://github.com/Aryetis/Obenkyobo/files/11795057/log.txt)
2. Yes, it shows the "This software is still in development" popup and I can't click the "ok" button.  
3. I changed the line, but it still gets stuck on the popup.
4. I installed the latest AIO 0.3.0 version.

**Comment by Aryetis** [comment]
Thanks for the log,
Ok so from what I can read there's no crash, error, freeze, etc... Meaning it is likely due to qt5-kobo-platform-plugin (the library I'm using to interface qt and kobo hardware) not interpreting touch inputs correctly. One of its usual bug is flipping the X or Y axis of the touch input for some devices.
Ok so here's the plan : 
1/ Try to mirror your touch inputs on the X/Y axis to see if that's the problem.
2/ Try this [dev-build ](https://hijackedbrain.com/data/Obenkyobo_AIO_Release_0.3.1.zip ) (there will be bugs, mostly in the "file explorer" section) with an updated version of qt5-kobo-platform-plugin. 
3/ If that still does not work, this weekend I'll try to update qt5-kobo-platform-plugin once more and add even more logs so I can figure out exactly what's going on.

**Comment by imoBooze** [comment]
Okay,
1. It still doesn't seem to be registering the taps.
2. The issue persists, though the font and button seem to be larger.
3. Thank you.

**Comment by Aryetis** [comment]
Necrobumping this thread.

As you can guess, things didn't go as planned. Last year I confirmed that the problem came from the qt5-kobo-platform-plugin and waited for the maintainer to fix it somehow. When it became clear that they were too busy/moved on I decided to give it a try and reinstalled everything from the ground up to get a clean workspace.... That was a huge mistake. It turns out that on top of the qt plugin, the toolchain setup scripts were also broken.... Without a proper way to compile anything I kinda gave up for some months. Sinking many many many hours trying to fix the whole thing across multiple work sessions. Until a couple weeks ago! I finally managed to fix the toolchain setup scripts, updated qt5-kobo-platform-plugin dependencies and compile Obenkyobo with it. 

I also caved in to the new Kobo colour hype and got a Kobo Libra Colour and can now replicate the issue on one of my device. So hopefully I'll manage to fix it. So far I can display stuff, including colours (although Red and Blue channel are inversed but that's a known FBInk issue that should get fixed (for real) soon™. But yes inputs are broken.

tldr : Wasn't my code that caused the issue. It's been nobody's code for a couple of years cause the person behind it moved on. I'm kinda taking back the torch on this code and trying to fix it. No promises.

**Comment by Aryetis** [comment]
> Okay,
> 
>     1. It still doesn't seem to be registering the taps.
> 
>     2. The issue persists, though the font and button seem to be larger.
> 
>     3. Thank you.

Okay, I've made [this debug version](https://hijackedbrain.com/data/Obenkyobo_0.3.1.Beta.zip) that works on my Kobo Libra Colour. Please let met know if this works for you. Please note that this is beta stuff so you won't have any support for wifi / screen settings / battery levels.
If not, please relaunch the application, tap the top left corner then top right corner, then bottom left, then bottom right. Shutdown your device and send the log.txt so I can check what's happenning with the touch inputs in more details.

**Comment by imoBooze** [comment]
It's working now!  I'm able to get past the welcome screen and use the app as normal.

**Comment by Aryetis** [comment]
Nice, glad to know.
Incoming "non beta" version should be incoming soon™ then. Probably a couple of weeks.
